### PR TITLE
chore(docs): Setting Up a Gatsby Site Without Gatsby New

### PR DESCRIPTION
**Pull request by mattfrankjames**
The purpose of this is to provide documentation for developers in jobs that have a private NPM clone, but do not have access to public Github repos.

## Description

Adding documentation to set up a new Gatsby project without the `gatsby new` command.

### CODE REVIEW

**Comment by freiksenet** [comment]
Thank you for you PR! You need to also add it to `www/src/data/sidebars/doc-links.yaml` for this doc to be discoverable on the sidebar. Also consider adding this to `https://www.gatsbyjs.org/docs/preparing-your-environment/` section.

**Comment by mattfrankjames** [comment]
Thank you for the feedback. I've incorporated your suggestions. I put the link in the "Using Gatsby Professionally" section of the sidebar, but I'd like your thoughts on that.

File: docs/docs/setting-up-gatsby-without-gatsby-new.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+## Setting Up a Gatsby Site Without Gatsby New
+
+There are many Enterprise level companies that maintain an internal clone of the NPM registry for security purposes. If you work for such a company, you may find that you are able to successfully run `npm install -g gatsby-cli` but cannot run the `gatsby new <project-source>` as the `gatsby new` command clones a repo from a public GitHub repository. Many companies block public GitHub, which will cause the `gatsby new` command to fail. Not to worry, though, you can set up a new Gatsby site without the `gatsby new` command with a few quick steps.
+
+## Preparing your environment
+To get started with Gatsby, you’ll need to make sure you have the following software tools installed:
+
+1. [Node.js](https://www.gatsbyjs.org/tutorial/part-zero/#install-nodejs)
```
Comment (LekoArts) [inline docs/docs/setting-up-gatsby-without-gatsby-new.md]:
```suggestion
1. [Node.js](/tutorial/part-zero/#install-nodejs)
```

File: docs/docs/setting-up-gatsby-without-gatsby-new.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+1. [npm CLI](https://www.gatsbyjs.org/tutorial/part-zero/#familiarize-with-npm)
```
Comment (LekoArts) [inline docs/docs/setting-up-gatsby-without-gatsby-new.md]:
```suggestion
1. [npm CLI](/tutorial/part-zero/#familiarize-with-npm)
```

File: docs/docs/setting-up-gatsby-without-gatsby-new.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+1. [Gatsby CLI](https://www.gatsbyjs.org/tutorial/part-zero/#install-the-gatsby-cli)
```
Comment (LekoArts) [inline docs/docs/setting-up-gatsby-without-gatsby-new.md]:
```suggestion
1. [Gatsby CLI](/tutorial/part-zero/#install-the-gatsby-cli)
```

File: docs/docs/setting-up-gatsby-without-gatsby-new.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+For step-by-step installation instructions and detailed explanations of the required software, head on over to the [Gatsby tutorial](https://www.gatsbyjs.org/tutorial/part-zero/).
```
Comment (LekoArts) [inline docs/docs/setting-up-gatsby-without-gatsby-new.md]:
```suggestion
For step-by-step installation instructions and detailed explanations of the required software, head on over to the [Gatsby tutorial](/tutorial/part-zero/).
```

File: docs/docs/setting-up-gatsby-without-gatsby-new.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+Next, you'll need to set up NPM within your project.
+```shell
+npm init
+```
+
+Fill out the prompts for the package.json file that is generated. If you'd like to skip that, you can run `npm init -y` and a pre-filled package.json will be generated for you.
+
+Now, you'll need to install the neccessary packages that Gatsby relies on to work its magic.
+```shell
+npm i gatsby react react-dom
```
Comment (LekoArts) [inline docs/docs/setting-up-gatsby-without-gatsby-new.md]: Being explicit here is better for beginners

```suggestion
npm install --save gatsby react react-dom
```

File: docs/docs/setting-up-gatsby-without-gatsby-new.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+```
+
+Once your packages have installed, you'll likely want to add some of the Gatsby specific files at the root of your project.
```
Comment (LekoArts) [inline docs/docs/setting-up-gatsby-without-gatsby-new.md]: I think we can remove this part as these files are not necessary to get a Gatsby site working

File: docs/docs/setting-up-gatsby-without-gatsby-new.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+```shell
+touch gatsby-config.js
+touch gatsby-browser.js
+```
+
+Next, you'll add a `src` directory and a `pages` directory inside of that.
```
Comment (LekoArts) [inline docs/docs/setting-up-gatsby-without-gatsby-new.md]:
```suggestion
Next, you'll add a `src` directory and a `pages` directory inside your project.
```

File: docs/docs/setting-up-gatsby-without-gatsby-new.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+);
+```
+
+Finally, go back to the root of your project and run the `gatsby develop` command to start a development server and begin coding.
+```shell
+cd ../../ 
+gatsby develop
+```
+
+And that's it! You should now have your initial page running on `localhost:8000` with a GraphiQL IDE running on `localhost:8000/___graphql`. From here, you can follow the rest of the [Gatsby tutorial](https://www.gatsbyjs.org/tutorial/part-zero/#set-up-a-code-editor) starting with setting up a code editor to get the full experience of what Gatsby can offer.
```
Comment (LekoArts) [inline docs/docs/setting-up-gatsby-without-gatsby-new.md]:
```suggestion
And that's it! You should now have your initial page running on `localhost:8000` with a GraphiQL IDE running on `localhost:8000/___graphql`. From here, you can follow the rest of the [Gatsby tutorial](/tutorial/part-zero/#set-up-a-code-editor) starting with setting up a code editor to get the full experience of what Gatsby can offer.  
```

File: docs/docs/setting-up-gatsby-without-gatsby-new.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+cd my-new-gatsby-site
+```
+
+Next, you'll need to set up NPM within your project.
+
+```shell
+npm init
+```
+
+Fill out the prompts for the package.json file that is generated. If you'd like to skip that, you can run `npm init -y` and a pre-filled package.json will be generated for you.
```
Comment (marcysutton) [inline docs/docs/setting-up-gatsby-without-gatsby-new.md]:
```suggestion
Fill out the prompts for the `package.json` file that is generated. If you'd like to skip that, you can run `npm init -y` and a pre-filled `package.json` will be generated for you.
```

File: docs/docs/setting-up-gatsby-without-gatsby-new.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+Now, you'll need to install the neccessary packages that Gatsby relies on to work its magic.
```
Comment (marcysutton) [inline docs/docs/setting-up-gatsby-without-gatsby-new.md]:
```suggestion
Now, you'll need to install the necessary packages that Gatsby relies on to work its magic.
```

File: docs/docs/setting-up-gatsby-without-gatsby-new.md [hunk 11]
```diff
@@ -0,0 +1,66 @@
+## Setting Up a Gatsby Site Without Gatsby New
```
Comment (marcysutton) [inline docs/docs/setting-up-gatsby-without-gatsby-new.md]:
```suggestion
## Setting Up a Gatsby Site Without `gatsby new`
```
@LekoArts can we use syntax highlighting in frontmatter?

File: www/src/data/sidebars/doc-links.yaml [hunk 12]
```diff
@@ -460,6 +460,8 @@
           link: /docs/convincing-others/
         - title: Your First Professional Project
           link: /docs/first-professional-project/
+        - title: Setting Up Gatsby Without Gatsby New
```
Comment (marcysutton) [inline www/src/data/sidebars/doc-links.yaml]: I think this is fine here, but we could add a link to it from the Preparing Your Environment page. (I don't think it fits very well in that section of the sidebar)

Comment (mattfrankjames) [reply www/src/data/sidebars/doc-links.yaml]: I was a bit conflicted about it. It feels like an odd fit for most of the sections, but I have run into people who faced the same issue on Stack Overflow, so it seemed like this was an ok place.

**Comment by mattfrankjames** [comment]
Incorporate suggestions from @marcysutton .

File: docs/docs/setting-up-gatsby-without-gatsby-new.md (hunk 11 above)
Comment (LekoArts) [reply docs/docs/setting-up-gatsby-without-gatsby-new.md]: Yep, should work

File: www/src/data/sidebars/doc-links.yaml (hunk 12 above)
Comment (marcysutton) [reply www/src/data/sidebars/doc-links.yaml]: I went ahead and added a link from the Preparing Your Environment page, along with a link to the Quick start!

**Pull request merged**